interfaces.go

```diff
@@ -77,6 +77,12 @@ type GetDBConnector interface {
 	GetDBConn() (*sql.DB, error)
 }
 
+// SQLConnector represents SQL db connector which takes into account the current
+// database context
+type SQLConnector interface {
```

Comment:
This is a good point, but since it expresses a similar meaning to GetDBConnector (returns SQL db connector), I feel that the current naming is not easy to distinguish between them. Can we make it clear in naming that this interface needs to provide context?

Reply:
Agree, I feel the same about the naming. Maybe:

```go
type GetDBConnectorWithContext interface {
	GetDBConnWithContext(db *DB) (*sql.DB, error)
}
```

Reply:
I think it's ok